tutorials/batch-approve-swap.md

```diff
@@ -0,0 +1,402 @@
+    function swapDevForDemoToken() payable public {
+        // Verify the contract has enough tokens for the requested amount
+        uint256 amountTobuy = msg.value;
+        uint256 dexBalance = token.balanceOf(address(this));
+        require(amountTobuy > 0, "You need to send some DEV");
+        require(amountTobuy <= dexBalance, "Not enough tokens in the reserve");
+        // If enough, swap the DEV to DTOKs
+        token.transfer(msg.sender, amountTobuy);
+        emit Bought(amountTobuy);
+    }
+
+    // Function to swap DTOK for DEV tokens
+    function swapDemoTokenForDev(uint256 amount) public {
+        // Make sure the requested amount is greater than 0 and the caller
+        // has approved the requested amount of tokens to be transferred
+        require(amount > 0, "You need to sell at least some tokens");
+        uint256 allowance = token.allowance(msg.sender, address(this));
+        require(allowance >= amount, "Check the token allowance");
+        // Transfer the DTOKs to the contract
+        token.transferFrom(msg.sender, address(this), amount);
+        // Transfer the DEV tokens back to the caller
+        payable(msg.sender).transfer(amount);
+        emit Sold(amount);
+    }
+}
+```
+
+In the `Batch.sol` file, you can paste in the [batch precompile contract](https://github.com/PureStake/moonbeam/blob/master/precompiles/batch/Batch.sol){target=_blank}. 
+
+### Compile & Deploy Contracts {: #compile-deploy-contracts }
+
+To compile the contracts, we'll go ahead and run the following Hardhat command:
+
+```
+npx hardhat compile
+```
+
+![Compile contracts](/images/tutorials/batch-approve-swap/batch-1.png)
+
+After compilation, an `artifacts` directory is created: it holds the bytecode and metadata of the contract, which are `.json` files. It’s a good idea to add this directory to the `.gitignore` file.
+
+Next, we can deploy the `SimpleDex` contract, which upon deployment will automatically deploy the `DemoToken` contract and mint 1000 DTOKs and assign half of them to the `SimpleDex` contract and the other half to the address that you're initiating the deployment from.
+
+We'll also add some initial liquidity to the contract by passing in a `value` when calling `deploy`. Since the value needs to be in Wei, we can use `ethers.utils.parseEther` to pass in a value such as `"0.5"` DEV and it will convert the value to Wei for us.
+
+Before deploying the contract, we'll need to create the deployment script. We'll create a new directory for the script and name it `scripts` and add a new file to it called `deploy.js`:
+
+```
+mkdir scripts && touch scripts/deploy.js
+```
+
+In the `deploy.js` script, you can paste in the following code, which will deploy the `SimpleDex` contract and print the address of the contract to the terminal upon successful deployment:
+
+```js
+async function main() {
+  // DEV liquidity to add
+  const value = ethers.utils.parseEther("INSERT-AMOUNT-OF-DEV");
```

Comment:
Potentially add a suggested value?

Reply:
I added a suggested value in the comment